source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -75,6 +77,9 @@ void SslSocket::setTransportSocketCallbacks(Network::TransportSocketCallbacks& c
   // Use custom BIO that reads from/writes to IoHandle
   BIO* bio = BIO_new_io_handle(&callbacks_->ioHandle());
   SSL_set_bio(rawSsl(), bio, bio);
+  SSL_set_ex_data(rawSsl(), ctx_->getSslExDataCBIdx(), static_cast<void*>(callbacks_));
+  SSL_set_ex_data(rawSsl(), ctx_->getSslExDataCfgIdx(),
```

Comment:
This shouldn't be needed. If you store the `SslSocket` instead of `callbacks_`, you can access a pointer to the context from the `SslSocket`. Also, lifetime issues are much more clear.

Reply:
will fix